test/Main.hs

```diff
@@ -0,0 +1,71 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE LambdaCase #-}
+-- |
+-- Module: Main
+-- Copyright: Copyright © 2024 Kadena LLC.
+-- License: MIT
+-- Maintainer: Lars Kuhtz <lars@kadena.io>
+-- Stability: experimental
+--
+module Main
+( main
+) where
+
+import Control.Monad
+
+import Data.Aeson
+
+import Foreign.C.String
+
+-- internal modules
+import PlonkVerify
+
+-- -------------------------------------------------------------------------- --
+-- main
+
+main :: IO ()
+main = do
+    example "fibonacci_fixture"
+    example "epoch_change"
+    example "inclusion_fixture"
+
+-- --------------------------------------------------------------------------
+-- Run Example
+
+example :: String -> IO ()
+example name = do
+    p <- readProof name
+    dataDir <- newCString "./assets"
+    proof <- newCString (_proofProof p)
+    vkeyHash <- newCString (_proofVKey p)
+    committedValuesDigest <- newCString (_proofPublicValues p)
+
+    rawRes <- verify_plonk_bn254
+        dataDir
+        proof
+        vkeyHash
+        committedValuesDigest
+
+    when (rawRes == 0) $ error "Verification failed"
```

Comment:
Just to double check: is it correct that 0 means success?

Out of curiosity: in case verification is not successful, is there a way to obtain details about the failure?

Reply:
> is it correct that 0 means success?

No, we're returning `0` for failure and `1` for success. We can change this easily [here](https://github.com/argumentcomputer/lurk-hs/blob/main/src/lib.rs#L58-L63)

> Out of curiosity: in case verification is not successful, is there a way to obtain details about the failure?

We can get and return the error message returned by the verifier, but that is not necessarily actionable or descriptive. (As you've seen by the "algebraic relation does not hold" error message)

Reply:
Oops, yeah, I meant "0 means failure". The code is already correct.